src/sentry/shared_integrations/client/proxy.py

```diff
@@ -117,7 +118,14 @@ def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest
 
         # E.g. client.get("/chat.postMessage") -> proxy_path = 'chat.postMessage'
         assert self.base_url and self.proxy_url
+
         base_url = self.base_url.rstrip("/")
+        if not prepared_request.url.startswith(base_url):
+            parsed = urlparse(prepared_request.url)
+            proxy_path = parsed.path
+            base_url = parsed._replace(path="", params="", query="", fragment="").geturl()
```

Comment:
Using private APIs of urllib3 isn't ideal. We could import `urllib.parse.ParseResult` and make a new instance

```python
base_url = ParseResult(scheme=base_url.scheme, netloc=base_url.netloc, path="", params="", query="", fragment="")
```

Reply:
@markstory Interestingly the documentation makes the recommendation to use `_replace()` with examples: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse